projects/core/src/site-context/facade/currency.service.ts

```diff
@@ -11,41 +15,42 @@ import {
   getActiveCurrency
 } from '../store/selectors/currencies.selectors';
 import { OccConfig } from '../../occ/config/occ-config';
-import { Currency } from '../../occ-models/occ.models';
 
 @Injectable()
 export class CurrencyService {
-  readonly currencies$: Observable<Currency[]> = this.store.pipe(
-    select(getAllCurrencies)
-  );
-
-  readonly activeCurrency$: Observable<string> = this.store.pipe(
-    select(getActiveCurrency)
-  );
-
   constructor(
     private store: Store<StateWithSiteContext>,
     private config: OccConfig
   ) {
-    this.initActiveCurrency();
-    this.loadCurrencies();
+    this.initActive();
+    this.load();
+  }
+
+  get(): Observable<Currency[]> {
+    return this.store.pipe(select(getAllCurrencies));
   }
 
-  protected loadCurrencies() {
+  load(): void {
```

Comment:
Do we want `load` and `initActive` to be `protected`? They should be only called in `construct`.

Reply:
I will make them `protected`